pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go

```diff
@@ -409,43 +409,20 @@ func doHTTPFilterListOperation(patchContext networking.EnvoyFilter_PatchContext,
 		doHTTPFilterOperation(patchContext, patches, listener, fc, filter, httpFilter, &httpFiltersRemoved)
 	}
 	for _, cp := range patches[networking.EnvoyFilter_HTTP_FILTER] {
+		// We should process INSERT_AFTER patches at the end so that we can apply them after other patches.
```

Comment:
Same rationale applies to insert-before, or anything else.
The real feature at hand is *dependency* between envoy filters.
And the solution we are looking for is to apply the dependencies *before* applying modifications.

I don’t think just fixing it for insert-after is Enough.

Reply:
Agreed. Just fixing it for insert-after is enough. But could not think of better way to handle both -  Do you have any suggestions?

Reply:
The current state is the older resource is applied first everything else being the same. In absence of this change the insert-after patch will fail since the relative anchor does not exist yet. 

Thinking out aloud, if a patch is rejected due to missing anchor, then we look thru the remaining patches that may add that anchor and apply that. If that patch is also relative to another anchor we search and apply that, and so on.

this way we do not change the current filter order unless the current processing is resulting in filter rejection. Changing filter order will be a breaking change.

I think this is a bit complicated, but correct. Wdyt?

Reply:
Yes. I thought about that approach as you mentioned it is a bit complicated. Let me attempt that and see where we go.

Reply:
Can we just rely on the order of EnvoyFilters? Just name the patches so that dependency comes before use.